### DISCUSSION TITLE
Brainstorming, how to mix Transaction DB with OLAP/DuckDB

### DISCUSSION THREAD

**Discussion by manticore-projects** [duckdb/duckdb#6973] [Q&A]
Greetings.

We run quite large transactional DBs (Oracle, Postgres, MS SQL Server) for financial valuation with more than 30 Mill records per table as per day.

The processes features 4 Steps: ETL/Import, Valuation, Accounting, Reporting.
The first 3 Steps fall under the "When not to use OLAP/DuckDB" category as there are millions of tiny parallel transactions.

Although the last Step "Reporting" looks very much like an OLAP usecase of aggregating large amounts of data ("total balance as per reporting dimension").

So, my question was: What would be the best way to marry our existing infrastructure with DuckDB? Was is to extract `Parquet` files of the result tables after Step 3 and import those into DuckDB and run the Analytics there? (I am worried, if the overhead of exporting/importing Parquet files was justified by the faster query execution).

Any better idea or "best practice"?

I also would like to ask for any argument to use the relatively new DuckDB over ClickHouse?

Thanks a lot in advance.

**Comment by orzom411** [comment]
Nice thing I’ve finding with DuckDB is no need to “import” the parquet files.  That is all my pain point right now is extraction from the transactional systems (many transactional systems).  That is, I am finding as a processing agent DuckDB is very well crafted for native parquet files.  Hope this helps with your use case.

**Comment by l1t1** [comment]
maybe a real-time synchronization from oltp to duckdb is needed

**Reply by manticore-projects** [reply]
I was thinking about the same. Instead of writing the results into the transactional table and exporting that table to a parquet file, I eventually may write to parquet directly.

Before I go that route, I will need to run some proper performance tests for my specific use-case so I am going to export the involved tables (via JSQLParser --> Table Names Finder plus JDBC Result MetaData for Indexes), import into DuckDB and the compare the query execution speed.
